src/tblib/__init__.py

```diff
@@ -19,22 +19,16 @@
 
 
 class _AttrDict(dict):
+    __slots__ = ()
+
     def __getattr__(self, attr):
         return self[attr]
 
 
-class __traceback_maker(Exception):
```

Comment:
I think my intention was to use a specific exception (instead of a generic one) because I wanted to aid debugging. Eg, if there's a failure somewhere and the user gets to see it then it's easy to search for it. Not so for the generic `Exception`.